### PR TITLE
Refactor: Split socket_spec.cr into separate files

### PR DESCRIPTION
**Pull request by straight-shoota**
This PR splits the large monolithic `socket_spec.cr` into separate files for the individual components. This makes it easier to follow the specific component specs in their own context and improves overall readability by segregating individual features.

This is the first in a series of PR's to improve the Socket API.

### CODE REVIEW

File: spec/std/socket/address_spec.cr [hunk 1]
```diff
@@ -161,3 +161,91 @@ describe Socket::UNIXAddress do
...
+    Socket.ip?(":ffff:c0a8:5e4").should be_false
+    Socket.ip?("0:0:0:0:0:ffff:c0a8:5e4").should be_true
+    Socket.ip?("0:0:0:0:ffff:c0a8:5e4").should be_false
+    Socket.ip?("0::ffff:c0a8:5e4").should be_true
+    Socket.ip?("::0::ffff:c0a8:5e4").should be_false
+    Socket.ip?("c0a8").should be_false
+  end
+
+  describe ".unix" do
+    it "creates a unix socket" do
```
Comment (bew) [inline spec/std/socket/address_spec.cr]: why this landed to `address_spec.cr` ?

File: spec/std/socket/address_spec.cr [hunk 2]
```diff
@@ -161,3 +161,91 @@ describe Socket::UNIXAddress do
...
+      sock.should be_a(Socket)
+      sock.family.should eq(Socket::Family::UNIX)
+      sock.type.should eq(Socket::Type::STREAM)
+
+      sock = Socket.unix(Socket::Type::DGRAM)
+      sock.type.should eq(Socket::Type::DGRAM)
+    end
+  end
+
+  it ".accept" do
```
Comment (bew) [inline spec/std/socket/address_spec.cr]: ditto

File: spec/std/socket/address_spec.cr (hunk 1 above)
Comment (bew) [reply spec/std/socket/address_spec.cr]: after looking at the rest, it looks like you forgot to remove these specs from this file, ?

File: spec/std/socket/address_spec.cr [hunk 3]
```diff
@@ -161,3 +161,91 @@ describe Socket::UNIXAddress do
     end
   end
 end
+
+describe Socket do
+  # Tests from libc-test:
+  # http://repo.or.cz/libc-test.git/blob/master:/src/functional/inet_pton.c
+  it ".ip?" do
+    # dotted-decimal notation
+    Socket.ip?("0.0.0.0").should be_true
```
Comment (RX14) [inline spec/std/socket/address_spec.cr]: Maybe this should be on `Socket::Address`.

**Comment by asterite** [comment]
All CIs failed with this:

```Failures:
  1) Socket .accept
bind: Address already in use (Errno)
  from src/socket.cr:73:15 in 'bind'
  ...

File: spec/std/socket/address_spec.cr (hunk 3 above)
Comment (straight-shoota) [reply spec/std/socket/address_spec.cr]: Yep, that's actually a separate commit not included in this PR.

File: spec/std/socket/address_spec.cr (hunk 1 above)
Comment (straight-shoota) [reply spec/std/socket/address_spec.cr]: Darn, looks like something went awry during rebase. I'll fix it.

**Comment by straight-shoota** [comment]
@RX14 yay, the issue with Unix socket path limitation with MacOS's temp dir is back: https://circleci.com/gh/crystal-lang/crystal/12013?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Pull request merged**